### PR TITLE
feat(feature_flags): Added inequality conditions

### PR DESCRIPTION
**Pull request by gwlester**
**Issue #701, if available:**

## Description of changes:

Added inequality conditions.

**Checklist**

 

* [ ] [Meet tenets criteria](https://awslabs.github.io/aws-lambda-powertools-python/#tenets)
* [x] Update tests
* [x] Update docs
* [x] PR title follows [conventional commit semantics](https://github.com/awslabs/aws-lambda-powertools-python/blob/376ec0a2ac0d2a40e0af5717bef42ff84ca0d1b9/.github/semantic.yml#L2)

## Breaking change checklist

**RFC issue #701**:

* [ ] Migration process documented
* [ ] Implement warnings (if it can live side by side)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

-----
[View rendered docs/utilities/feature_flags.md](https://github.com/gwlester/aws-lambda-powertools-python/blob/feature-701/docs/utilities/feature_flags.md)

### CODE REVIEW

**Comment by gwlester** [comment]
@heitorlessa  -- conflicts resolved.

**Comment by gwlester** [comment]
@heitorlessa -- tests fixed.

File: tests/functional/feature_flags/test_feature_flags.py [hunk 1]
```diff
@@ -802,3 +802,338 @@ def test_get_configuration_with_envelope_and_raw(mocker, config):
...
+                    ],
+                }
+            },
+        }
+    }
+    feature_flags = init_feature_flags(mocker, mocked_app_config_schema, config)
+    toggle = feature_flags.evaluate(name="my_feature", context={"tenant_id": "345345435", "username": "a"}, default=False)
+    assert toggle == expected_value
+
+def test_flags_not_eqaul_match(mocker, config):
```
Comment (heitorlessa) [inline tests/functional/feature_flags/test_feature_flags.py:834]:
```suggestion
def test_flags_not_equal_match(mocker, config):
```

File: tests/functional/feature_flags/test_feature_flags.py [hunk 2]
```diff
@@ -802,3 +802,338 @@ def test_get_configuration_with_envelope_and_raw(mocker, config):
 
     assert "log_level" in config
     assert "log_level" not in features_config
+    
+##
+## Inequality test cases
+##
+
+# Test not equals
+def test_flags_not_eqaul_no_match(mocker, config):
```
Comment (heitorlessa) [inline tests/functional/feature_flags/test_feature_flags.py:811]:
```suggestion
def test_flags_not_equal_no_match(mocker, config):
```

File: tests/functional/feature_flags/test_feature_flags.py [hunk 3]
```diff
@@ -802,3 +802,338 @@ def test_get_configuration_with_envelope_and_raw(mocker, config):
...
+                            "value": "2021.10.31",
+                        }
+                    ],
+                }
+            },
+        }
+    }
+    feature_flags = init_feature_flags(mocker, mocked_app_config_schema, config)
+    toggle = feature_flags.evaluate(name="my_feature", context={"tenant_id": "345345435", "username": "a", "current_date": "2021.10.31"}, default=False)
+    assert toggle == expected_value
```
Comment (heitorlessa) [inline tests/functional/feature_flags/test_feature_flags.py:903]: perhaps merge them in a single test w/ different features?

**Review by heitorlessa — CHANGES_REQUESTED** [review]
Thanks a lot again @gwlester - my only ask would be to merge these `_1` and `_2` tests so we reduce the number of imports that pytests have to make

File: tests/functional/feature_flags/test_feature_flags.py (hunk 3 above)
Comment (gwlester) [reply tests/functional/feature_flags/test_feature_flags.py:903]: We could, but I always lean toward one test for one condition.

We were just bit by one test that was supposed to test several conditions and due to the complexity was not testing them but was still appearing to pass.

**Comment by heitorlessa** [comment]
Fair enough

**Review by heitorlessa — APPROVED** [review]

**Pull request merged**